cli/README.md

```diff
@@ -0,0 +1,135 @@
+## Cook CLI
+
+### Installation
+
+To install the Cook CLI, clone this repo and from this folder run:
+
+```bash
+python3 setup.py install
+```
+
+This will install the `cs` command on your system.
+
+### Configuration
+
+The Cook CLI client is designed with "end user ergonomics" in mind. 
+Custom defaults may be provided for all commands. 
+Multiple clusters are supported via configuration.
+
+In order to use the Cook CLI, you’ll need a configuration file. 
+`cs` looks first for a `.cs.json` file in the current directory, and then for a `.cs.json` file in your home directory. 
+The path to this file may also be provided manually via the command line with the `--config` option.
+
+There is a sample `.cs.json` file included in this directory, which looks something like this:
+
+```json
+{
+  "defaults":{
+    "submit": {
+      "mem": 128,
+      "cpus": 1
+    }
+  },
+  "clusters": [
+    {
+      "name": "dev0",
+      "url": "http://127.0.0.1:12321/"
+    },
+    {
+      "name": "dev1",
+      "url": "http://127.0.0.1:12322/"
+    }
+  ]
+}
+```
+
+The `defaults` map contains default values that are passed each time a command is run. 
+For instance, the `submit` map above is used to set the defaults for `mem` and `cpus` when calling the `submit` command. 
+These can always be overridden directly from the command line. 
+For most commands, all available clusters are tried in order. 
+Each entry in the `clusters` array conforms to a cluster specification ("spec"). 
+A cluster spec requires a name and a url pointing to a Cook Scheduler.
+
+### Commands
+
+The fastest way to learn more about `cs` is with the `-h` (or `--help`) option.
+
+All global options (`--cluster`, `--config`, etc.) can be provided when using subcommands.
+
+#### `submit`
+
+You can submit one or more jobs with `submit`. 
+A single command can be provided as a positional argument, or multiple commands can be provided, via `stdin`. 
+Jobs can also be passed as "raw" JSON data by setting the `--raw` flag. 
+The `submit` command returns UUIDs, one per line, for each created job.
+
+#### `wait`
+
+You can wait for jobs, job instances, and job groups to complete with `wait`. 
+UUIDs are passed as positional arguments.
+
+#### `show`
+
+You can query jobs, instances, and groups with `show`. 
+UUIDs are passed as positional arguments. 
+This command shows information about the queried jobs, instances, and groups. 
+If you use `--json`, the command returns a JSON representation instead of tables.
+
+### Examples
+
+Simple job creation:
+```shell
+$ cs submit echo 1
+Attempting to submit on dev0 cluster...
+Job submission succeeded on dev0. Your job UUID is:
+9ee74508-8b07-4b9d-88b4-9af33f8f77a0
+```
+
+Create multiple jobs, each with their own command:
+```shell
+$ printf 'echo 1\necho 2' | cs submit
```

Comment:
I took a random sampling of 10 production jobs, and based on those, added `test_complex_commands`, which has two separate scenarios that can be handled with `cs`. I didn't add a scenario for all 10 because there was some overlap and some were trivial.

Multiline commands can be submitted like this, and are supported by the cook scheduler:

```bash
$ cs submit $'echo Hello\necho World'
Attempting to submit on dev0 cluster...
Job submission succeeded on dev0. Your job UUID is:
9dd44ef1-4ada-46ae-826b-b3150f43b487

$ cs show 9dd44ef1-4ada-46ae-826b-b3150f43b487

=== Job: 9dd44ef1-4ada-46ae-826b-b3150f43b487 (dpo_job) ===

Cluster   dev0                 Attempts    1 / 1
Memory    128 MB               Job Status  Success
CPUs      1.0                  Submitted   just now
User      root                 
Priority  50                   

Command:
echo Hello
echo World

Job Instance                          Run Time                      Host        Instance Status
5544bcc1-633f-4b83-a70d-009e5eab67a2  0 seconds (started just now)  172.17.0.6  Success
```

Reply:
Nice.  I believe also just <pre>cs submit 'echo Hello
echo World' would work.

Reply:
You are correct, I tested that as well and it worked